[quant] Add quantized::conv1d op benchmarck

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38449 [quant] Support for functional quantized::conv1d
* #38441 [quant][graphmode] Add support for quantized conv1d + relu fusion
* #38438 [quant] Fusion support for conv1d + ReLU
* #38407 [quant] Return default qconfig when backend is 'none'
* #38341 [quant][graphmode] Add quantized::conv1d to graphmode
* **#38332 [quant] Add quantized::conv1d op benchmarck**
* #38283 [quant] Add support for Quantized Conv1d and ConvRELU1d
* #38248 [quant] Add support for quantized::conv1d operator
* #38247 [quant] Implement unsqueeze/squeeze for per-channel qtensor

Summary:

Test Plan:

python -m pt.qconv_test --test QConv1d_N1_IC128_OC256_L64_G1_kernel3_stride1_pad0
Forward Execution Time (us) : 147.844

python -m pt.conv_test --test Conv1d_IC128_OC256_kernel3_stride1_N1_L64_cpu
Forward Execution Time (us) : 470.750

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21553662](https://our.internmc.facebook.com/intern/diff/D21553662)

supriyar committed:
[quant] Add quantized::conv1d op benchmarck
Summary:

Test Plan:

python -m pt.qconv_test --test QConv1d_N1_IC128_OC256_L64_G1_kernel3_stride1_pad0
Forward Execution Time (us) : 147.844

python -m pt.conv_test --test Conv1d_IC128_OC256_kernel3_stride1_N1_L64_cpu
Forward Execution Time (us) : 470.750

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on benchmarks/operator_benchmark/pt/qconv_test.py:
```diff
@@ -58,6 +70,29 @@
     tags=["resnext101_32x4d"],
 )
 
+class QConv1dBenchmark(op_bench.TorchBenchmarkBase):
```
can we factor out the common part of this class so that future support is easier? e.g. we might want conv3d benchmark as well.

supriyar committed:
Update on "[quant] Add quantized::conv1d op benchmarck"


Summary:

Test Plan:

python -m pt.qconv_test --test QConv1d_N1_IC128_OC256_L64_G1_kernel3_stride1_pad0
Forward Execution Time (us) : 147.844

python -m pt.conv_test --test Conv1d_IC128_OC256_kernel3_stride1_N1_L64_cpu
Forward Execution Time (us) : 470.750

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 reviewed (approved):
Thanks
